apis/apps/v1alpha1/componentdefinition_types.go

```diff
@@ -565,18 +565,20 @@ type ComponentVolume struct {
 
 // ReplicasLimit defines the valid range of number of replicas supported.
 //
-// +kubebuilder:validation:XValidation:rule="self.minReplicas >= 0 && self.maxReplicas <= 128",message="the minimum and maximum limit of replicas should be in the range of [0, 128]"
+// +kubebuilder:validation:XValidation:rule="self.minReplicas >= 0 && self.maxReplicas <= 2147483647",message="the minimum and maximum limit of replicas should be in the range of [0, math.MaxInt32]"
 // +kubebuilder:validation:XValidation:rule="self.minReplicas <= self.maxReplicas",message="the minimum replicas limit should be no greater than the maximum"
 type ReplicasLimit struct {
 	// The minimum limit of replicas.
 	//
-	// +kubebuilder:validation:Required
-	MinReplicas int32 `json:"minReplicas"`
+	// +kubebuilder:default=1
```

Comment:
Default 0 is a better choice when you scale a component replicas to 0 and get limit validation errors. Then you will have to update the component spec to admit such case. Anyway, default min 1 has too much implicit assumptions inside.

Reply:
Scale-in operations will delete PVCs. 

Setting the minimum replica number to 1 is intended to prevent accidental deletion and data loss.